src/tools/tidy/src/deps.rs

```diff
@@ -29,47 +33,425 @@ static LICENSES: &'static [&'static str] = &[
+//    ("scopeguard", "0.1.2"),
+//    ("scopeguard", "0.3.3"),
+//    ("semver", "0.6.0"),
+//    ("semver", "0.8.0"),
+//    ("semver", "0.9.0"),
+//    ("semver-parser", "0.7.0"),
+//    ("serde", "1.0.27"),
+//    ("serde_derive", "1.0.27"),
+//    ("serde_derive_internals", "0.19.0"),
+//    ("serde_ignored", "0.0.4"),
+//    ("serde_json", "1.0.9"),
+//    ("serialize", "0.0.0"),
+//    ("shared_child", "0.2.1"),
+//    ("shell-escape", "0.1.3"),
+//    ("shlex", "0.1.1"),
+//    ("smallvec", "0.6.0"),
+//    ("socket2", "0.3.0"),
+//    ("stable_deref_trait", "1.0.0"),
+//    ("std", "0.0.0"),
+//    ("std_unicode", "0.0.0"),
+//    ("strsim", "0.6.0"),
+//    ("syn", "0.11.11"),
+//    ("synom", "0.11.3"),
+//    ("synstructure", "0.6.1"),
+//    ("syntax", "0.0.0"),
+//    ("syntax_ext", "0.0.0"),
+//    ("syntax_pos", "0.0.0"),
+//    ("syntex_errors", "0.52.0"),
+//    ("syntex_pos", "0.52.0"),
+//    ("syntex_syntax", "0.52.0"),
+//    ("tar", "0.4.14"),
+//    ("tempdir", "0.3.5"),
+//    ("term", "0.0.0"),
+//    ("term", "0.4.6"),
+//    ("termcolor", "0.3.3"),
+//    ("termion", "1.5.1"),
+//    ("test", "0.0.0"),
+//    ("textwrap", "0.9.0"),
+//    ("thread-id", "2.0.0"),
+//    ("thread_local", "0.2.7"),
+//    ("thread_local", "0.3.5"),
+//    ("tidy", "0.1.0"),
+//    ("time", "0.1.39"),
+//    ("toml", "0.2.1"),
+//    ("toml", "0.4.5"),
+//    ("toml-query", "0.6.0"),
+//    ("unicode-bidi", "0.3.4"),
+//    ("unicode-normalization", "0.1.5"),
+//    ("unicode-segmentation", "1.2.0"),
+//    ("unicode-width", "0.1.4"),
+//    ("unicode-xid", "0.0.3"),
+//    ("unicode-xid", "0.0.4"),
+//    ("unreachable", "1.0.0"),
+//    ("unstable-book-gen", "0.1.0"),
+//    ("unwind", "0.0.0"),
+//    ("url", "1.6.0"),
+//    ("url_serde", "0.2.0"),
+//    ("userenv-sys", "0.2.0"),
+//    ("utf8-ranges", "0.1.3"),
+//    ("utf8-ranges", "1.0.0"),
+//    ("vcpkg", "0.2.2"),
+//    ("vec_map", "0.8.0"),
+//    ("void", "1.0.2"),
+//    ("walkdir", "1.0.7"),
+//    ("walkdir", "2.0.1"),
+//    ("winapi", "0.2.8"),
+//    ("winapi", "0.3.4"),
+//    ("winapi-build", "0.1.1"),
+//    ("winapi-i686-pc-windows-gnu", "0.4.0"),
+//    ("winapi-x86_64-pc-windows-gnu", "0.4.0"),
+//    ("wincolor", "0.1.4"),
+//    ("workspace_symbol", "0.1.0"),
+//    ("ws2_32-sys", "0.2.1"),
+//    ("xattr", "0.1.11"),
+//    ("xz2", "0.1.4"),
+//    ("yaml-rust", "0.3.5"),
 ];
 
+// Some types for Serde to deserialize the output of `cargo metadata` to...
+
+#[derive(Deserialize)]
+struct Output {
+    packages: Vec<Package>,
+
+    // Not used, but needed to not confuse serde :P
+    #[allow(dead_code)] resolve: Resolve,
+}
+
+#[derive(Deserialize)]
+struct Package {
+    name: String,
+    version: String,
+
+    // Not used, but needed to not confuse serde :P
+    #[allow(dead_code)] id: String,
+    #[allow(dead_code)] source: Option<String>,
+    #[allow(dead_code)] manifest_path: String,
+}
+
+// Not used, but needed to not confuse serde :P
```

Comment:
Hm if these aren't used, are they needed? I think serde should ignore unused fields by default?

Reply:
Hmm... When I tried that it was giving me deserialization errors. I will take another look.